src/smt/seq_unicode.cpp

```diff
@@ -84,50 +84,47 @@ namespace smt {
     }
 
     bool seq_unicode::try_bound(theory_var v, unsigned min, unsigned max) {
-        push_scope();
+
+        local_scope sc(*this);
         dl.init_var(v);
         theory_var v0 = ensure0();
+
         bool sat = dl.enable_edge(dl.add_edge(v, v0, s_integer(-1 * min), null_literal));
-        if (!sat) {
-            pop_scope(1);
-            return false;
-        }
+        if (!sat) return false;
+
         sat = dl.enable_edge(dl.add_edge(v0, v, s_integer(max), null_literal));
-        if (!sat) {
-            pop_scope(1);
-            return false;
-        }
-        pop_scope(1);
+        if (!sat) return false;
+
         return true;
     }
 
     bool seq_unicode::try_assignment(theory_var v, unsigned ch) {
         return try_bound(v, ch, ch);
     }
 
-    void seq_unicode::try_make_nice(svector<theory_var> char_vars) {
+    void seq_unicode::try_make_nice(svector<theory_var> const& char_vars) {
 
         // Try for each character
         unsigned i = 0;
         for (auto v : char_vars) {
 
             // Skip character constants, since they can't be changed
-            if (seq.is_any_const_char(th.get_expr(v))) continue;
+            if (seq.is_const_char(th.get_expr(v))) continue;
 
             // Check if the value is already nice
             int val = get_value(v);
             if (val <= NICE_MAX && val >= NICE_MIN) continue;
 
             // Try assigning the variable to a nice value
             if (try_bound(v, NICE_MIN, NICE_MAX)) {
-                try_assignment(v, NICE_MIN + i);
+                try_assignment(v, NICE_MIN + (i % (NICE_MAX - NICE_MIN)));
```

Comment:
I made a mistake in the review it should have been i % (NICE_MAX - NICE_MIN + 1)